packages/core/src/contrib/ConstraintsUtils.ts

```diff
@@ -132,47 +132,35 @@ export const overlappingCircleLine = (
 };
 
 /**
- * Require that shape `s1` overlaps shape `s2` with some padding `padding`.
+ * Require that shape `s1` is at a distance of `distance` from shape `s2`.
  */
-export const overlappingAABBsMinkowski = (
+export const atDistLabel = (
   [t1, s1]: [string, any],
   [t2, s2]: [string, any],
-  padding: VarAD = constOf(0.0)
+  distance: VarAD
 ): VarAD => {
-  const box1 = bboxFromShape([t1, s1]);
-  const box2 = bboxFromShape([t2, s2]);
-  const [pc1, pc2] = rectangleDifference(box1, box2, padding);
-  const [xp, yp] = ops.vmul(constOf(0.5), ops.vadd(pc1, pc2));
-  const [xr, yr] = ops.vmul(constOf(0.5), ops.vsub(pc2, pc1));
-  const [xq, yq] = ops.vsub([absVal(xp), absVal(yp)], [xr, yr]);
-  const e1 = sqrt(
-    add(
-      constOf(10e-15),
-      add(
-        squared(max(sub(xp, xr), constOf(0.0))),
-        squared(max(sub(yp, yr), constOf(0.0)))
-      )
-    )
+  let pt;
+  if (shapedefs[t1].isLinelike) {
+    // Position label close to the arrow's end
+    pt = s1.end.contents;
+  } else {
+    // Only assume shape1 has a center
+    pt = shapeCenter([t1, s1]);
+  }
+
+  // Get polygon of text (box)
+  // TODO: Make this a GPI property
```

Comment:
Sorry, this is an old piece of code. I deleted this function in #751, but had to put it in because of backward compatibility issues. Still would like to delete this in the future. I should document the changes better next time.